consul/src/main/scala/io/buoyant/consul/v1/AgentApi.scala

```diff
@@ -0,0 +1,31 @@
+package io.buoyant.consul.v1
+
+import com.twitter.conversions.time._
+import com.twitter.finagle.http
+import com.twitter.finagle.service.Backoff
+import com.twitter.finagle.stats.{DefaultStatsReceiver, StatsReceiver}
+import com.twitter.util.{Closable, Duration, Future}
+
+object AgentApi {
+  def apply(c: Client): AgentApi = new AgentApi(c, s"/$versionString")
+}
+
+class AgentApi(
+  override val client: Client,
+  override val uriPrefix: String,
+  override val backoffs: Stream[Duration] = Backoff.exponentialJittered(1.milliseconds, 5.seconds),
+  override val stats: StatsReceiver = DefaultStatsReceiver
+) extends BaseApi with Closable {
+
+  val agentPrefix = s"$uriPrefix/agent"
+
+  // https://www.consul.io/docs/agent/http/agent.html#agent_self
+  def self(retry: Boolean = false): Future[Self] = {
+    val req = mkreq(http.Method.Get, s"$agentPrefix/self")
+    executeJson[Self](req, retry).map(_.value)
+  }
+}
+
+case class Config(Domain: Option[String])
+
+case class Self(Config: Option[Config])
```

Comment:
Regarding case of field names - here I followed example from https://github.com/BuoyantIO/linkerd/blob/master/consul/src/main/scala/io/buoyant/consul/v1/CatalogApi.scala#L62 - aren't field names should match JSON field names for parser to work properly? Or it can figure out the thing? Because in Consul JSON output they're starting from upper case.